File: Homework 1/environment variables/environment-variables.c
/*
   This program demonstrates a simple use of environment variables
   (and also argc and argv).
*/
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char* argv[])
{
   int i;
   char* value;

   // walk the argv array and look for environment variables
   for (i = 1; i < argc; i++)
   {
      value = getenv( argv[i] );
      if ( ! value )
      {
         printf( "%s is not an enviromnent variable.\n", argv[i] );
      }
      else
      {
         printf( "The value of %s is %s.\n", argv[i], value );
      }
   }

   return 0;
}//main

File: Homework 1/environment variables/EnvironmentVariables.java
/*
   A simple demonstration of using environment variables
   (and command line arguments).

   See
      http://docs.oracle.com/javase/6/docs/api/java/lang/System.html#getenv(java.lang.String)
*/

public class EnvironmentVariables
{
   public static void main(String[] args)
   {
      // walk the args array and look for environment variables
      for (int i = 0; i < args.length; i++)
      {
         String value = System.getenv( args[i] );
         if ( value == null )
         {
            System.out.println( args[i] + " is not an environment variable." );
         }
         else
         {
            System.out.println( "The value of " + args[i] + " is " + value );
         }
      }
   }//main()
}